### PR TITLE
fix all loops in alias graph rather than just first order loops

### PR DESCRIPTION
**Pull request by oscardssmith**
This is the followup to #2137 that is able to detect loops of higher orders.

### CODE REVIEW

File: src/structural_transformation/pantelides.jl [hunk 1]
```diff
@@ -121,25 +121,33 @@ function computed_highest_diff_variables(structure, ag::Union{AliasGraph, Nothin
...
-                        # If we have a self-loop in the stem, we could have the
-                        # var′ also alias to the original stem. In that case, the
-                        # derivative of the stem is highest differentiated, because of the loop
+                # If we have a self-loop in the stem, we could have the
+                # var′ also alias to the original stem. In that case, the
+                # derivative of the stem is highest differentiated, because of the loop
+                loop_found = false
+                var′ = invview(var_to_diff)[var]
+                while var′ !== nothing
+                    if var′ === stem || haskey(ag, var′) && ag[var′][2] == stem
```
Comment (YingboMa) [inline src/structural_transformation/pantelides.jl:130]: Maybe add a pair of parenthesis here. Mix || and && is pretty confusion.

**Review by YingboMa — COMMENTED** [review]
Do you mean `ÿ` instead of ` ̈y`? It should be `y\ddot<tab>`.

File: src/structural_transformation/pantelides.jl [hunk 2]
```diff
@@ -121,25 +121,33 @@ function computed_highest_diff_variables(structure, ag::Union{AliasGraph, Nothin
...
-                        # If we have a self-loop in the stem, we could have the
-                        # var′ also alias to the original stem. In that case, the
-                        # derivative of the stem is highest differentiated, because of the loop
+                # If we have a self-loop in the stem, we could have the
+                # var′ also alias to the original stem. In that case, the
+                # derivative of the stem is highest differentiated, because of the loop
+                loop_found = false
+                var′ = invview(var_to_diff)[var]
+                while var′ !== nothing
+                    if var′ === stem || (haskey(ag, var′) && ag[var′][2] == stem)
```
Comment (YingboMa) [inline src/structural_transformation/pantelides.jl:130]: It should also be `var′ === stem` right?

Comment (oscardssmith) [reply src/structural_transformation/pantelides.jl:130]: I don't think it matters. They're both `Int`, right?

Comment (YingboMa) [reply src/structural_transformation/pantelides.jl:130]: It's a good way to get silent bugs when refactoring.

Comment (oscardssmith) [reply src/structural_transformation/pantelides.jl:130]: fair enough.

Comment (YingboMa) [inline src/structural_transformation/pantelides.jl:130]:
```suggestion
                    if var′ == stem || (haskey(ag, var′) && ag[var′][2] == stem)
```

**Pull request merged**